.github/workflows/release.yaml

```diff
@@ -0,0 +1,280 @@
+        run: |
+          fullVersion="unstable-${{ inputs.sha }}"
+          aliasedVersion="unstable"
+
+          echo "full-version=$fullVersion" >> "$GITHUB_OUTPUT"
+          echo "aliased-version=$aliasedVersion" >> "$GITHUB_OUTPUT"
+      - id: testing
+        if: ${{ inputs.kind == 'testing' }}
+        run: |
+          fullVersion="testing-${{ inputs.sha }}"
+          aliasedVersion=""
+
+          echo "full-version=$fullVersion" >> "$GITHUB_OUTPUT"
+          echo "aliased-version=$aliasedVersion" >> "$GITHUB_OUTPUT"
+
+  draft-full-version-release:
+    runs-on: ubuntu-latest
+    needs: extract-version-details
+    permissions:
+      contents: write
+      id-token: write
+    env:
+      GH_TOKEN: ${{ github.token }}
+    steps:
+      - uses: actions/checkout@v3
+        with:
+          ref: ${{ inputs.sha }}
+      - name: Ensure this release has not been published
+        run: gh release view ${{ needs.extract-version-details.outputs.full-version }} --json isDraft | jq -e '.isDraft == true'
+      - name: Push full version tags
+        run: |
+          git tag --force ${{ needs.extract-version-details.outputs.full-version }}
+          git push --tags --force origin ${{ needs.extract-version-details.outputs.full-version }}
+      - id: create-full-version-release
+        continue-on-error: true
+        name: Create the full version release
+        run: |
+          gh release create ${{ needs.extract-version-details.outputs.full-version }} \
+            --draft \
+            --prerelease=${{ inputs.kind != 'production' }} \
+            --generate-notes \
+            --title "Captain ${{ needs.extract-version-details.outputs.full-version }}"
+      - name: Update the release if it already exists
+        if: ${{ steps.create-full-version-release.outcome == 'failure' }}
+        run: gh release edit ${{ needs.extract-version-details.outputs.full-version }} --tag ${{ needs.extract-version-details.outputs.full-version }}
+
+  build-and-upload-binaries:
+    runs-on: ubuntu-latest
+    needs:
+      - extract-version-details
+      - draft-full-version-release
+    permissions:
+      contents: write
+      id-token: write
+    strategy:
+      matrix:
+        os: [linux, darwin, windows]
+        arch: [amd64, arm64]
+    env:
+      GH_TOKEN: ${{ github.token }}
+    steps:
+      - uses: actions/checkout@v3
+        with:
+          ref: ${{ needs.extract-version-details.outputs.full-version }}
+      - uses: cachix/install-nix-action@v18
+        with:
+          extra_nix_config: |
+            keep-derivations = true
+            keep-outputs = true
+      - run: |
+          GOOS=${{ matrix.os }} \
+            GOARCH=${{ matrix.arch }} \
+            CGO_ENABLED=0 \
+            LDFLAGS="-w -s -X github.com/rwx-research/captain-cli.Version=${{ needs.extract-version-details.outputs.full-version }}" \
+            nix develop --command mage
+      - if: ${{ matrix.os != 'windows' }}
```

Comment:
💭 maybe worth merging these if the only difference is the binary name

Reply:
This was yanked from the old script. I'll take a look